R/label-compose.R

```diff
@@ -0,0 +1,48 @@
+#' Compose two or more label formatters together
+#'
+#' This labeller provides a general mechanism for composing two or more
+#' labellers together.
+#'
+#' @param ... One or more labelling functions. These will be applied to breaks
+#'   consecutively.
+#'   [Lambda syntax][rlang::as_function] is allowed.
+#' @param call A call to display in error messages.
+#'
+#' @return A labelling function that applies the provided
+#'   functions to breaks to return labels.
+#'
+#' @export
+#'
+#' @examples
+#' demo_continuous(
+#'   c(-100, 100),
+#'   labels = compose_label(abs, number, ~paste0(.x, " foobar"), toupper)
+#' )
+#'
+#' # Same result
+#' demo_continuous(
+#'   c(-100, 100),
+#'   labels = compose_label(abs, label_number(suffix = " FOOBAR"))
+#' )
+compose_label <- function(..., call = caller_env()) {
+
+  label_list <- list2(...)
+  if (length(label_list) == 0) {
+    return(identity)
+  }
+  label_list <- lapply(label_list, as_function, call = call)
+
+  function(x) {
+    if (length(x) == 0) {
+      return(character())
+    }
+    orig <- x
+    for (labeller in label_list) {
+      x <- labeller(x)
+      attr(x, "orig") <- orig
```

Comment:
Attaching the original breaks as an attribute here so that 'aware' function could use original numeric breaks if they've already been converted to character, or something of the sort.

Reply:
maybe something more specific than `orig`? `orig_label` is at least more descriptive

Reply:
I prefer `orig_breaks` then because we attach breaks and not labels as attributes. This confusion just go to show how `orig` was ambiguous 😅